Gill_Kyle/index.py

```diff
@@ -0,0 +1,46 @@
+from django.conf import settings
+from django_mako_plus import view_function, jscontext
+from django.shortcuts import render
+from catalog import models as cmod
+import math
+
+
+@view_function
+def process_request(request, category_id, current_page: int = 1):
+    selected_category = cmod.Category.objects.filter(id=category_id)
+    if selected_category.count() > 0:
+        page_title = selected_category[0].name
+        print(page_title)
+        print("found selected category")
+        products = cmod.Product.objects.filter(category=category_id)
+        number_of_pages = math.ceil(products.count() / 6)
+    else:
+        page_title = 'Products'
+        print("didn't find category")
+        products = cmod.Product.objects.all()
+        number_of_pages = math.ceil(products.count() / 6)
+
+    print(page_title)
+    context = {
+        'page_title': page_title,
+        'products': products[:6],
```

Comment:
The products[:6] just finds the 6th product right?

Reply:
This will actually find everything _up to_ the 6th product 🙂 these are what I have in my recently viewed array.